SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialWorkerConnection.cpp

```diff
@@ -418,3 +410,43 @@ void USpatialWorkerConnection::CacheWorkerAttributes()
 		CachedWorkerAttributes.Add(UTF8_TO_TCHAR(Attributes->attributes[Index]));
 	}
 }
+
+USpatialNetDriver* USpatialWorkerConnection::GetSpatialNetDriverChecked() const
+{
+	auto NetDriver = Cast<USpatialNetDriver>(World->GetNetDriver());
+	checkf(NetDriver, TEXT("SpatialNetDriver was invalid during SpatialOS connection event!"));
+	return NetDriver;
+}
+
+
+// TODO: UNR-962 - move connection events in to native connection handling codepath (eg, UEngine::HandleNetworkFailure)
+void USpatialWorkerConnection::OnConnectionSuccess()
+{
+	bIsConnected = true;
+	GetSpatialNetDriverChecked()->HandleOnConnected();
+}
+
+void USpatialWorkerConnection::OnPreConnectionFailure(const FString& Reason)
+{
+	bIsConnected = false;
+	GetSpatialNetDriverChecked()->HandleOnConnectionFailed(Reason);
+}
+
+void USpatialWorkerConnection::OnConnectionFailure()
+{
+	bIsConnected = false;
+
+	Worker_OpList* OpList = Worker_Connection_GetOpList(WorkerConnection, 0);
+	for (int i = 0; i < static_cast<int>(OpList->op_count); i++)
```

Comment:
The other place we iterate over the OpList, we use a size_t as the type for `i`. Could we do the same here just to be consistent?